bundles/com.espressif.idf.launch.serial.core/src/com/espressif/idf/launch/serial/internal/SerialFlashLaunchConfigDelegate.java

```diff
@@ -117,10 +117,12 @@ protected void launchInternal(ILaunchConfiguration configuration, String mode, I
 	{
 		IStringVariableManager varManager = VariablesPlugin.getDefault().getStringVariableManager();
 
-		String location = varManager.performStringSubstitution(IDFUtil.getIDFPythonEnvPath());
+		String location = configuration.getAttribute(ICDTLaunchConfigurationConstants.ATTR_LOCATION,
+				SYSTEM_PATH_PYTHON);
+		location = varManager.performStringSubstitution(location);
 		if (StringUtil.isEmpty(location))
 		{
-			location = configuration.getAttribute(ICDTLaunchConfigurationConstants.ATTR_LOCATION, SYSTEM_PATH_PYTHON);
+			location = varManager.performStringSubstitution(IDFUtil.getIDFPythonEnvPath());
 		}
```

Comment:
theoretically, we shouldn't get there, because it's not allowed to save empty locations now. Mb it's worth deleting this part.